src/brouwers/shop/api/viewsets.py

```diff
@@ -0,0 +1,56 @@
+from rest_framework import views, viewsets
+from rest_framework.response import Response
+
+from ..models import Cart, CartProduct, Product
+from .filters import CartProductFilter
+from .serializers import (
+    CartSerializer, ProductSerializer, ReadCartProductSerializer,
+    WriteCartProductSerializer
+)
+
+
+class ProductViewSet(viewsets.ModelViewSet):
+    queryset = Product.objects.all()
+    serializer_class = ProductSerializer
+
+
+class CartViewSet(views.APIView):
+    def get(self, request, *args, **kwargs):
+        cart_id = None
+
+        if request.session.get('cart_id'):
+            cart_id = request.session['cart_id']
+        elif hasattr(request.user, 'carts'):
+            cart = request.user.carts.open().last()
+
+            if cart:
+                cart_id = cart.id
+
+        if cart_id:
+            cart = Cart.objects.get(id=cart_id)
+        else:
+            if request.user.is_authenticated():
```

Comment:
```suggestion
            if request.user.is_authenticated:
```
this is no longer a method in django 1.11 and higher